platform/default/src/mbgl/storage/online_file_source.cpp

```diff
@@ -170,7 +170,15 @@ class OnlineFileSource::Impl {
 private:
 
     void networkIsReachableAgain() {
-        for (auto& request : allRequests) {
+        // Notify requests about network reachability in order of their priority.
+        std::vector<decltype(allRequests)::value_type> requests;
+        requests.reserve(allRequests.size());
+        requests.insert(requests.end(), allRequests.begin(), allRequests.end());
+        std::sort(requests.begin(), requests.end(), [](auto& a, auto& b) {
+            return a->resource.priority < b->resource.priority;
+        });
+
+        for (auto& request : requests) {
```

Comment:
Rather than copying + sorting them, we could just do two passes over the loop and notify the high priority ones first, then the regular priority ones.

Reply:
done. thanks.